File: ruby/fizzbuzz.rb
#!/usr/bin/env ruby

fizzbuzz = Proc.new do |i|
  if i % 3 == 0 and i % 5 == 0
    "Fizzbuzz"
  elsif i % 3 == 0
    "Fizz"
  elsif i % 5 == 0
    "Buzz"
  else
    i.to_s
  end
end

(1..100).map(&fizzbuzz).each { |i| print i + "\n" }

File: scala/fizzbuzz.scala
#!/bin/sh
exec scala $0 $@
!#
for (i <- 1 until 101)
  if ((i % 3 == 0) && (i % 5 == 0)) {
    println("Fizzbuzz");
  } else if (i % 3 == 0) {
    println("Fizz");
  } else if (i % 5 == 0) {
    println("Buzz");
  } else {
    println(i);
  }
